engine/src/main/java/com/datatorrent/stram/plan/physical/PTOperator.java

```diff
@@ -85,6 +85,7 @@
     private static final long serialVersionUID = 201312112033L;
 
     public final LogicalPlan.StreamMeta logicalStream;
+    public Locality locality;
```

Comment:
StreamMeta is shared with the upstream operator, so changing that has unwanted side effects. I am exploring the options to create a new StreamMeta object for each Unifiers.